sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponse.java

```diff
@@ -4,34 +4,75 @@
 package com.generic.core.http.okhttp.implementation;
 
 import com.generic.core.http.models.HttpRequest;
+import com.generic.core.http.models.HttpResponse;
 import com.generic.core.models.BinaryData;
+import com.generic.core.models.HeaderName;
+import okhttp3.Headers;
 import okhttp3.Response;
 import okhttp3.ResponseBody;
 
 /**
- * Default HTTP response for OkHttp.
+ * Base response class for OkHttp with implementations for response metadata.
  */
-public final class OkHttpResponse extends OkHttpResponseBase {
-    // Previously, this was 4096, but it is being changed to 8192 as that more closely aligns to what Netty uses as a
-    // default and will reduce the number of small allocations we'll need to make.
-    private static final int BYTE_BUFFER_CHUNK_SIZE = 8192;
-
+public class OkHttpResponse extends HttpResponse {
     private final ResponseBody responseBody;
 
-    public OkHttpResponse(Response response, HttpRequest request, boolean eagerlyConvertHeaders) {
-        super(response, request, eagerlyConvertHeaders);
+    private BinaryData value;
+
+    public OkHttpResponse(Response response, HttpRequest request, boolean eagerlyConvertHeaders, byte[] bodyBytes) {
+        super(request, response.code(), eagerlyConvertHeaders ? fromOkHttpHeaders(response.headers())
+            : new OkHttpToAzureCoreHttpHeadersWrapper(response.headers()), null);
 
         // innerResponse.body() getter will not return null for server returned responses.
         // It can be null:
         // [a]. if response is built manually with null body (e.g for mocking)
         // [b]. for the cases described here
         // [ref](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/).
         this.responseBody = response.body();
+        this.value = bodyBytes == null ? null : BinaryData.fromBytes(bodyBytes);
+    }
+
+    /**
+     * Creates {@link com.generic.core.models.Headers Generic Core's headers} from {@link Headers OkHttp headers}.
+     *
+     * @param okHttpHeaders {@link Headers OkHttp headers}.
+     *
+     * @return {@link com.generic.core.models.Headers Generic Core's headers}.
+     */
+    static com.generic.core.models.Headers fromOkHttpHeaders(Headers okHttpHeaders) {
```

Comment:
I checked the `generic-sdk-core-2` branch and this is still there in `[OkHttpResponse](https://github.com/Azure/azure-sdk-for-java/blob/c618cfa17f19e9845c50c30476952b6c1b957907/sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponseBase.java#L51)`. If this was removed from Azure Core we can follow suit in a subsequent PR.

Reply:
The copy paste removed the deprecation warning in the original file https://github.com/Azure/azure-sdk-for-java/blob/feature/generic-sdk-core-2/sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponseBase.java#L50

We should not be carrying forward azure-core deprecated methods to generic-core. 
Could you add back the deprecation and do a follow up PR to remove it?

Reply:
The deprecation suppression was there originally because [in Azure Core](https://github.com/Azure/azure-sdk-for-java/blob/e0a615cac870157c72b66b9a1b5b20cb6d78a153/sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpAsyncResponseBase.java#L84) the deprecated method [HttpHeaders.add()](https://github.com/Azure/azure-sdk-for-java/blob/e0a615cac870157c72b66b9a1b5b20cb6d78a153/sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java#L102) is used and I forgot to remove it when bringing in the class to Generic Core. We don't use the deprecated method in this version of Core so there's no need to suppress a warning that's not popping up.